nixos/modules/services/mail/isync.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.isync;
+in {
+
+  options.services.isync = {
+    enable = mkEnableOption "Isync, a software to dispose your mailbox(es) as a local Maildir(s)";
+
+    install = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to install a user service for Isync. Once
+        the service is started, emails will be fetched automatically.
+
+        The service must be manually started for each user with
+        "systemctl --user start isync" or globally through
+        <varname>services.isync.enable</varname>.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.isync;
+      defaultText = "pkgs.isync";
+      description = "Isync derivation to use.";
+    };
+
+    path = mkOption {
+      type = types.listOf types.path;
+      default = [];
+      example = literalExample "[ pkgs.pass pkgs.bash pkgs.notmuch ]";
+      description = "List of derivations to put in Isync's path.";
+    };
+
+    onCalendar = mkOption {
+      type = types.str;
+      default = "*:0/3"; # every 3 minutes
+      description = "How often is isync started. Default is '*:0/3' meaning every 3 minutes. See systemd.time(7) for more information about the format.";
+    };
+
+    timeoutStartSec = mkOption {
+      type = types.str;
+      default = "120sec"; # Kill if still alive after 2 minutes
+      description = "How long waiting for isync before killing it. Default is '120sec' meaning every 2 minutes. See systemd.time(7) for more information about the format.";
+    };
+  };
+  config = mkIf (cfg.enable || cfg.install) {
+    systemd.user.services.isync = {
+      description = "Isync: a software to dispose your mailbox(es) as a local Maildir(s)";
+      serviceConfig = {
+        Type      = "oneshot";
+        ExecStart = "${cfg.package}/bin/mbsync -Va";
+        TimeoutStartSec = cfg.timeoutStartSec;
```

Comment:
Like this, this runs as `root`. See if `DynamicUser = true;` works

Reply:
with `DynamicUser = true;" produces ther error
`isync.service: Failed at step USER spawning /nix/store/bwfv5j8x9fbxnn5hy4y50yink1vn6hkz-isync-1.3.0/bin/>

Reply:
Try figuring out why this doesn't work, maybe check what DynamicUser does in `man systemd.exec`, look at the source of isync. It might be a really simple fix.

Reply:
It's a systemd user service (`systemd.user.services`, the additional `user` is unfortunately easy to miss), therefore it already runs as the "current" user instead of root. `DynamicUser` doesn't work as it needs to run as the user that's synchronizing the mailbox to access the home directory..

Reply:
Ahh it's a user service, I missed that indeed! Ignore this then